### DISCUSSION TITLE
Tests for mint tools

### DISCUSSION THREAD

**Discussion by devid8642** [linuxmint/discussions#8] [Development] [answered]
I'm a beginner developer and was looking into the possibility of adding tests to the mint-common project. But browsing through the other mint tools projects I realized that none of them had any tests. Is it a project decision or should I go ahead and think about testing?

**Comment by clefebvre** [comment, marked as answer]
It depends on what you want to actually test.

We don't care about metrics or code coverage. If you identify a potential for regression though and want to make sure a test would flag it, then you're welcome to add it. If it's something that's unlikely to regress though, then a test doesn't help.

Most identified regressions are part of our QA. I was recently working on xfce4-screensaver for instance... it breaks two tests of ours. Xubuntu 22.04/23.04 wouldn't have been released if we had been testing it. So we do take tests very seriously, but only if they bring something tangible to the table. The counterpart is that manual testing takes time, and automated testing means more code and a heavier framework around the code, i.e. more maintenance and less flexibility.

So in brief, testing for testing no. Testing a particular area for a very known reason, yes, absolutely.
